tests/unit/agent_main_test.py

```diff
@@ -182,68 +186,81 @@ def test_skipped_bytes_warnings(self):
-        config.scalyr_server = "foo.bar.com"
-        config.server_attributes = {"serverHost": "test"}
-        config.additional_file_paths = []
-        config.compression_level = 1
-        config.copying_manager_stats_log_interval = 60
-        config.parsed_max_send_rate_enforcement = 12345
-
-        config.verify_server_certificate = True
-        config.ca_cert_path = "/tmp/doesnt.exist"
-
-        platform_controller = PlatformController()
-        platform_controller.get_usage_info = AgentMainTestCase.fake_get_useage_info
-        agent = ScalyrAgent(platform_controller)
-        agent._ScalyrAgent__config = config
-        agent._ScalyrAgent__scalyr_client = agent._ScalyrAgent__create_client()
-
-        base_stats = agent_status.OverallStats()
-        base_stats.total_bytes_skipped = 500
-        test = agent._ScalyrAgent__calculate_overall_stats(
-            base_stats, copy_manager_warnings=True
-        )
-        self.assertIsNotNone(test)
-        self.assertLogFileContainsLineRegex(
-            ".*"
-            + re.escape(
-                "Warning, skipping copying log lines.  Only copied 0.0 MB/s log bytes when 0.0 MB/s were generated over the last 1.0 minutes. This may be due to max_send_rate_enforcement. Log upload has been delayed 0.0 seconds in the last 1.0 minutes  This may be desired (due to excessive bytes from a problematic log file).  Please contact support@scalyr.com for additional help."
-            )
-        )
-
-        config = mock.Mock()
-        config.scalyr_server = "foo.bar.com"
-        config.server_attributes = {"serverHost": "test"}
-        config.additional_file_paths = []
-        config.compression_level = 1
-        config.copying_manager_stats_log_interval = 60
-        config.parsed_max_send_rate_enforcement = None
-
-        config.verify_server_certificate = True
-        config.ca_cert_path = "/tmp/doesnt.exist"
-
-        platform_controller = PlatformController()
-        platform_controller.get_usage_info = AgentMainTestCase.fake_get_useage_info
-        agent = ScalyrAgent(platform_controller)
-        agent._ScalyrAgent__config = config
-        agent._ScalyrAgent__scalyr_client = agent._ScalyrAgent__create_client()
-
-        base_stats.total_bytes_skipped = 1000
-        test = agent._ScalyrAgent__calculate_overall_stats(
-            base_stats, copy_manager_warnings=True
-        )
-        self.assertIsNotNone(test)
-        with open(self.agent_log_path, "r") as f:
-            print((f.read()))
-        self.assertLogFileContainsLineRegex(
-            ".*"
-            + re.escape(
-                "Warning, skipping copying log lines.  Only copied 0.0 MB/s log bytes when 0.0 MB/s were generated over the last 1.0 minutes.  This may be desired (due to excessive bytes from a problematic log file).  Please contact support@scalyr.com for additional help."
-            )
-        )
+        with mock.patch("scalyr_agent.__scalyr__.INSTALL_TYPE", __scalyr__.DEV_INSTALL):
```

Comment:
Test case if refactored to work without the client session object in the ScalyrAgent instance.